src/docfx/lib/moniker/EvaluatorWithMonikersVisitor.cs

```diff
@@ -10,55 +10,52 @@ namespace Microsoft.Docs.Build
     internal class EvaluatorWithMonikersVisitor
     {
         private readonly Dictionary<string, List<Moniker>> _productMoniker;
-        private readonly Dictionary<string, Moniker> _monikerMap;
 
         public EvaluatorWithMonikersVisitor(MonikerDefinitionModel monikerDefinition)
         {
-            _monikerMap = monikerDefinition.Monikers.ToDictionary(x => x.MonikerName, StringComparer.OrdinalIgnoreCase);
+            MonikerMap = monikerDefinition.Monikers.ToDictionary(x => x.MonikerName, StringComparer.OrdinalIgnoreCase);
             _productMoniker = monikerDefinition.Monikers.GroupBy(x => x.ProductName).ToDictionary(g => g.Key, g => g.OrderBy(x => x.Order).ToList());
             MonikerOrder = SetMonikerOrder();
         }
 
+        public Dictionary<string, Moniker> MonikerMap { get; }
+
         public Dictionary<string, (string productName, int orderInProduct)> MonikerOrder { get; }
 
-        public IEnumerable<Moniker> Visit(ComparatorExpression expression)
+        public (List<Error>, IEnumerable<Moniker>) Visit(ComparatorExpression expression, SourceInfo<string?> monikerRange)
```

Comment:
does this need to be a list? or just a Error?

Reply:
It can just be `Error` here, but it needs to be `List<Error>` in the caller. So, maybe it would be easier to be the same signature. I am fine with both.

Reply:
perfer to use `Error?` if it is a big change, use the `addifNotNull` in the caller?

---
In reply to: [404512434](https://github.com/dotnet/docfx/pull/5704#discussion_r404512434) [](ancestors = 404512434)